### ISSUE TITLE
Suggestion: Read-only files

### ISSUE THREAD

**Issue by alphapapa** [orgzly/orgzly-android#91]
I saw in a comment on a PR that Orgzly generates Org files from a database when writing out changes.  I assume it writes the entire file at once, meaning that any differences in whitespace, etc. caused by Orgzly's own implementation (compared to org-mode) will cause parts of the file to change that haven't had any actual data changed.

Since I keep my Org files in a git repo, I like when the only changes in commits are actual data changes, not "noise" like whitespace.  So since this is how Orgzly works, I'd like to be able to use one file for writing to with Orgzly, and keep all the others read-only so that Orgzly won't modify them.  

Would it be possible to mark certain files to be synced read-only?

Thanks.

**Comment by nevenz** [comment]
> I saw in a comment on a PR that Orgzly generates Org files from a database when writing out changes.

Only if you modified the notebook from Orgzly or the linked file has been changed (its modification time or revision, if latter is available for the used repo type).

> I assume it writes the entire file at once, meaning that any differences in whitespace, etc. caused by Orgzly's own implementation (compared to org-mode) will cause parts of the file to change that haven't had any actual data changed.

Are you seeing this? Note content shouldn't change at all, everything else should be configurable.

> Would it be possible to mark certain files to be synced read-only?

There were suggestions before to add a flag for each notebook to skip it when syncing. I guess it could be multi-value instead (skip, load-only, save-only, sync).

**Comment by alphapapa** [comment]
> Only if you modified the notebook from Orgzly or the linked file has been changed (its modification time or revision, if latter is available for the used repo type).

I see, thanks.

> Are you seeing this? Note content shouldn't change at all, everything else should be configurable.

I haven't tested it, so I don't know.  I've only just started using Orgzly a little bit.  Sorry if I'm wasting your time here.  :)

So do you mean that Orgzly only reformats the entries in an Org file that have changed, and the rest of the file remains the same?  If so that's very cool, and I think that addresses my concern.

**Comment by nevenz** [comment]
> So do you mean that Orgzly only reformats the entries in an Org file that have changed, and the rest of the file remains the same? If so that's very cool, and I think that addresses my concern.

No, entire file is *generated*, but whitespace in content (text between headlines that's not planning times) shouldn't be modified.

See "Generated Org files" on http://www.orgzly.com/help. Both bullets there should be removed before next version though, as both will be configurable.

You can open a new issue with a specific problem you're seeing. This one could be kept for mentioned per-notebook sync option (though it kinda went in another direction, so you can close it too).

**Comment by nevenz** [comment]
Going to close this due to previously (by mail) reported (and now opened) #92.

But again, if you see any white-space issues, do open a new issue, thanks!

**Comment by alphapapa** [comment]
Well, I would still like to have an option to sync some files read-only, or one-way sync, or however it might be termed.  I don't think #92 is the same idea.  Could you leave this issue open as a potential feature?

**Comment by nevenz** [comment]
Wouldn't both be covered by:

> There were suggestions before to add a flag for each notebook to skip it when syncing. I guess it could be multi-value instead (skip, load-only, save-only, sync).

Sure, I'll reopen this one, but I suspect both issues will be handled together.
